### PR TITLE
fix(notification): incorrent status of DLM task notification

### LINKED ISSUE #3591: [Bug]: The archiving task failed to execute, but the message was sent successfully
### ODC version

ODC432

### OB version

independent

### What happened?

The archiving task failed to execute, but the message was sent successfully
image

### What did you expect to happen?

。

### How can we reproduce it (as minimally and precisely as possible)?

。

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:

We've added a hook for notification in `JobTerminateNotifyListener` , and determine the type of notification to be sent based on the status in `job_job` table.
But for DLM tasks , the job status is Incorrect. The real status should be found in `schedule_task`.

#### Which issue(s) this PR fixes:

Fixes #3591

### CODE REVIEW

**Review by guowl3 — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobTerminateNotifyListener.java [hunk 1]
```diff
@@ -60,7 +63,11 @@ public void onEvent(JobTerminateEvent event) {
             JobEntity jobEntity = taskFrameworkService.find(event.getJi().getId());
             scheduleTaskService.findByJobId(jobEntity.getId())
                     .ifPresent(task -> {
-                        broker.enqueueEvent(event.getStatus() == JobStatus.DONE ? eventBuilder.ofSucceededTask(task)
+                        TaskStatus status =
+                                "DLM".equalsIgnoreCase(jobEntity.getJobType())
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobTerminateNotifyListener.java:67]: Is there any consistent definition of jobType, string constant is not easy typo

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobTerminateNotifyListener.java [hunk 2]
```diff
@@ -60,7 +63,11 @@ public void onEvent(JobTerminateEvent event) {
...
+                                        ? dlmService.getFinalTaskStatus(task.getId())
+                                        : jobEntity.getStatus().convertTaskStatus();
+                        broker.enqueueEvent(status == TaskStatus.DONE ? eventBuilder.ofSucceededTask(task)
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobTerminateNotifyListener.java:70]: is cancel a candidate state?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobTerminateNotifyListener.java [hunk 3]
```diff
@@ -60,7 +63,11 @@ public void onEvent(JobTerminateEvent event) {
             JobEntity jobEntity = taskFrameworkService.find(event.getJi().getId());
             scheduleTaskService.findByJobId(jobEntity.getId())
                     .ifPresent(task -> {
-                        broker.enqueueEvent(event.getStatus() == JobStatus.DONE ? eventBuilder.ofSucceededTask(task)
+                        TaskStatus status =
+                                "DLM".equalsIgnoreCase(jobEntity.getJobType())
+                                        ? dlmService.getFinalTaskStatus(task.getId())
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/JobTerminateNotifyListener.java:68]: why dml maintains it's own status and may not same with job status?

**Pull request merged**